File: src/Java/myservicebus/src/main/java/com/myservicebus/JobConsumerOptions.java
package com.myservicebus;

import java.time.Duration;
import java.util.function.Consumer;

public final class JobConsumerOptions {
    private Duration jobTimeout = Duration.ofMinutes(30);
    private int concurrentJobLimit = 1;
    private int retryCount;
    private Duration retryDelay;
    private String jobTypeName;

    public JobConsumerOptions setJobTimeout(Duration timeout) {
        if (timeout == null || timeout.isZero() || timeout.isNegative()) {
            throw new IllegalArgumentException("timeout must be greater than zero");
        }
        jobTimeout = timeout;
        return this;
    }

    public JobConsumerOptions setConcurrentJobLimit(int limit) {
        if (limit <= 0) {
            throw new IllegalArgumentException("limit must be greater than zero");
        }
        concurrentJobLimit = limit;
        return this;
    }

    public JobConsumerOptions setRetry(Consumer<RetryConfigurator> configure) {
        if (configure == null) {
            throw new IllegalArgumentException("configure must not be null");
        }
        RetryConfigurator retry = new RetryConfigurator();
        configure.accept(retry);
        if (retry.getRetryCount() < 0) {
            throw new IllegalArgumentException("retry count must not be negative");
        }
        if (retry.getDelay() != null && retry.getDelay().isNegative()) {
            throw new IllegalArgumentException("retry delay must not be negative");
        }
        retryCount = retry.getRetryCount();
        retryDelay = retry.getDelay();
        return this;
    }

    public JobConsumerOptions setJobTypeName(String name) {
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("name must not be blank");
        }
        jobTypeName = name.trim();
        return this;
    }

    public Duration getJobTimeout() {
        return jobTimeout;
    }

    public int getConcurrentJobLimit() {
        return concurrentJobLimit;
    }

    public int getRetryCount() {
        return retryCount;
    }

    public Duration getRetryDelay() {
        return retryDelay;
    }

    public String getJobTypeName() {
        return jobTypeName;
    }
}

File: src/MyServiceBus/JobConsumerOptions.cs
namespace MyServiceBus;

public sealed class JobConsumerOptions
{
    public TimeSpan JobTimeout { get; private set; } = TimeSpan.FromMinutes(30);

    public int ConcurrentJobLimit { get; private set; } = 1;

    public int RetryCount { get; private set; }

    public TimeSpan? RetryDelay { get; private set; }

    public string? JobTypeName { get; private set; }

    public JobConsumerOptions SetJobTimeout(TimeSpan timeout)
    {
        if (timeout <= TimeSpan.Zero)
            throw new ArgumentOutOfRangeException(nameof(timeout), "The job timeout must be greater than zero.");

        JobTimeout = timeout;
        return this;
    }

    public JobConsumerOptions SetConcurrentJobLimit(int limit)
    {
        if (limit <= 0)
            throw new ArgumentOutOfRangeException(nameof(limit), "The concurrent job limit must be greater than zero.");

        ConcurrentJobLimit = limit;
        return this;
    }

    public JobConsumerOptions SetRetry(Action<RetryConfigurator> configure)
    {
        ArgumentNullException.ThrowIfNull(configure);
        var retry = new RetryConfigurator();
        configure(retry);
        if (retry.RetryCount < 0)
            throw new ArgumentOutOfRangeException(nameof(configure), "The retry count cannot be negative.");
        if (retry.Delay is { } delay && delay < TimeSpan.Zero)
            throw new ArgumentOutOfRangeException(nameof(configure), "The retry delay cannot be negative.");
        RetryCount = retry.RetryCount;
        RetryDelay = retry.Delay;
        return this;
    }

    public JobConsumerOptions SetJobTypeName(string name)
    {
        ArgumentException.ThrowIfNullOrWhiteSpace(name);
        JobTypeName = name.Trim();
        return this;
    }
}
